lib/ansible/parsing/vault/__init__.py

```diff
@@ -657,7 +657,13 @@ def decrypt(self, vaulttext, filename=None):
         :returns: a byte string containing the decrypted data and the vault-id that was used
 
         '''
-        plaintext, vault_id = self.decrypt_and_get_vault_id(vaulttext, filename=filename)
+        try:
+            plaintext, vault_id = self.decrypt_and_get_vault_id(vaulttext, filename=filename)
+        except AnsibleVaultError as e:
+            if C.ERROR_ON_VAULT_FAIL:
+                raise
+            else:
+                display.warning('Vault decryption failed: %s' % to_native(e))
```

Comment:
Now I'm wondering, what will be the value of `plaintext` in this scenario, and what the implications of that will be.